contracts/contracts/TACoApplication.sol

```diff
@@ -420,12 +470,45 @@ contract TACoApplication is
         if (!info.operatorConfirmed) {
             return info.tReward;
         }
-        uint256 result = (uint256(info.authorized) * (rewardPerToken() - info.rewardPerTokenPaid)) /
+        uint96 authorized = effectiveAuthorized(info.authorized, info);
+        uint256 result = (uint256(authorized) * (rewardPerToken() - info.rewardPerTokenPaid)) /
             FLOATING_POINT_DIVISOR +
             info.tReward;
         return result.toUint96();
     }
 
+    function effectiveAuthorized(
+        uint96 _authorized,
+        uint192 _penaltyPercent
+    ) internal pure returns (uint96) {
+        return uint96((_authorized * (PENALTY_BASE - _penaltyPercent)) / PENALTY_BASE);
+    }
+
+    /// @dev This view should be called after updateReward modifier
+    function effectiveAuthorized(
+        uint96 _authorized,
+        StakingProviderInfo storage _info
+    ) internal view returns (uint96) {
+        if (_info.endPenalty == 0) {
+            return _info.authorized;
+        }
+        return effectiveAuthorized(_authorized, _info.penaltyPercent);
+    }
+
+    /// @dev This view should be called after updateReward modifier
```

Comment:
What do you mean with these @dev comments?

Reply:
`updateReward` modifier has `resetReward` inside, so this view will give proper value after that check. Otherwise it will show penalty in cases when they may be reseted

Reply:
Right, this is related to my previous comment. What about something like:
```
/// @dev In case that a penalty period already ended, this view method may produce
///      outdated results if the penalty hasn't been reset, either by calling 
///      `resetReward` explicitly or any function with the `updateReward` modifier.
```

Reply:
✔️